commands/pull_request.go

```diff
@@ -222,6 +232,18 @@ func pullRequest(cmd *Command, args *Args) {
 		utils.Check(fmt.Errorf("Aborting due to empty pull request title"))
 	}
 
+	if flagPullRequestPush {
+		if remote == nil {
+			utils.Check(fmt.Errorf("Can't find remote for %s", head))
+		}
+
+		if args.Noop {
+			args.Before(fmt.Sprintf("Would push to %s/%s", remote.Name, head), "")
+		} else {
+			git.Push(remote.Name, "HEAD", head)
```

Comment:
There's something we need to take care of after pushing the branch. If we proceeded to make an API request immediately to try to create a pull request for a branch that was just pushed, the API might return an error that said `head` branch doesn't exist. This is because there is a little bit of delay between pushing a branch and GitHub registering that this branch exist right now. This is because GitHub processes git pushes via background jobs.

The tricky part is determining how long does this delay need to be. Maybe start with 5 seconds? In my experience, several seconds is usually all that's needed for GitHub to process it.

A more robust approach would be to try the API POST request, and if it's a 422 with `head` mentioned in validation errors, retry the POST automatically after several seconds, then again, and ultimately fail after a few tries. This might be tricky. Do you think it would be worth it?